### ISSUE TITLE
2.3.0: "Cannot determine the module for class AppComponent"

### ISSUE THREAD

**Issue by buu700** [angular/angular#13296]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

```
Compilation failedError: Unexpected value 'ChatCyphertext' declared by the module 'AppModule'
    at /usr/lib/node_modules/@angular/compiler/bundles/compiler.umd.js:17758:42
    at Array.forEach (native)
    at CompileMetadataResolver.getNgModuleMetadata (/usr/lib/node_modules/@angular/compiler/bundles/compiler.umd.js:17740:56)
    at addNgModule (/usr/lib/node_modules/@angular/compiler/bundles/compiler.umd.js:24285:60)
    at /usr/lib/node_modules/@angular/compiler/bundles/compiler.umd.js:24296:16
    at Array.forEach (native)
    at _createNgModules (/usr/lib/node_modules/@angular/compiler/bundles/compiler.umd.js:24295:28)
    at analyzeNgModules (/usr/lib/node_modules/@angular/compiler/bundles/compiler.umd.js:24174:16)
    at analyzeAndValidateNgModules (/usr/lib/node_modules/@angular/compiler/bundles/compiler.umd.js:24184:37)
    at AotCompiler.compileAll (/usr/lib/node_modules/@angular/compiler/bundles/compiler.umd.js:23923:20)
Compilation failedError: Cannot determine the module for class ChatCyphertext in /home/gibson/.build/shared/js/cyph/ui/components/chatcyphertext.ts!
Cannot determine the module for class ChatMain in /home/gibson/.build/shared/js/cyph/ui/components/chatmain.ts!
Cannot determine the module for class ChatMessageBox in /home/gibson/.build/shared/js/cyph/ui/components/chatmessagebox.ts!
Cannot determine the module for class Checkout in /home/gibson/.build/shared/js/cyph/ui/components/checkout.ts!
Cannot determine the module for class Contact in /home/gibson/.build/shared/js/cyph/ui/components/contact.ts!
Cannot determine the module for class FileInput in /home/gibson/.build/shared/js/cyph/ui/components/fileinput.ts!
Cannot determine the module for class Help in /home/gibson/.build/shared/js/cyph/ui/components/help.ts!
Cannot determine the module for class LinkConnection in /home/gibson/.build/shared/js/cyph/ui/components/linkconnection.ts!
Cannot determine the module for class Markdown in /home/gibson/.build/shared/js/cyph/ui/components/markdown.ts!
Cannot determine the module for class MdButton in /home/gibson/.build/shared/js/cyph/ui/components/material/mdbutton.ts!
Cannot determine the module for class MdCard in /home/gibson/.build/shared/js/cyph/ui/components/material/mdcard.ts!
Cannot determine the module for class MdCardContent in /home/gibson/.build/shared/js/cyph/ui/components/material/mdcardcontent.ts!
Cannot determine the module for class MdCardHeader in /home/gibson/.build/shared/js/cyph/ui/components/material/mdcardheader.ts!
Cannot determine the module for class MdCardHeaderText in /home/gibson/.build/shared/js/cyph/ui/components/material/mdcardheadertext.ts!
Cannot determine the module for class MdCardTitle in /home/gibson/.build/shared/js/cyph/ui/components/material/mdcardtitle.ts!
Cannot determine the module for class MdCardTitleText in /home/gibson/.build/shared/js/cyph/ui/components/material/mdcardtitletext.ts!
Cannot determine the module for class MdContent in /home/gibson/.build/shared/js/cyph/ui/components/material/mdcontent.ts!
Cannot determine the module for class MdFabSpeedDial in /home/gibson/.build/shared/js/cyph/ui/components/material/mdfabspeeddial.ts!
Cannot determine the module for class MdIcon in /home/gibson/.build/shared/js/cyph/ui/components/material/mdicon.ts!
Cannot determine the module for class MdInput in /home/gibson/.build/shared/js/cyph/ui/components/material/mdinput.ts!
Cannot determine the module for class MdList in /home/gibson/.build/shared/js/cyph/ui/components/material/mdlist.ts!
Cannot determine the module for class MdListItem in /home/gibson/.build/shared/js/cyph/ui/components/material/mdlistitem.ts!
Cannot determine the module for class MdMenu in /home/gibson/.build/shared/js/cyph/ui/components/material/mdmenu.ts!
Cannot determine the module for class MdOption in /home/gibson/.build/shared/js/cyph/ui/components/material/mdoption.ts!
Cannot determine the module for class MdProgressCircular in /home/gibson/.build/shared/js/cyph/ui/components/material/mdprogresscircular.ts!
Cannot determine the module for class MdProgressLinear in /home/gibson/.build/shared/js/cyph/ui/components/material/mdprogresslinear.ts!
Cannot determine the module for class MdSelect in /home/gibson/.build/shared/js/cyph/ui/components/material/mdselect.ts!
Cannot determine the module for class MdSidenav in /home/gibson/.build/shared/js/cyph/ui/components/material/mdsidenav.ts!
Cannot determine the module for class MdSlider in /home/gibson/.build/shared/js/cyph/ui/components/material/mdslider.ts!
Cannot determine the module for class MdSubheader in /home/gibson/.build/shared/js/cyph/ui/components/material/mdsubheader.ts!
Cannot determine the module for class MdSwitch in /home/gibson/.build/shared/js/cyph/ui/components/material/mdswitch.ts!
Cannot determine the module for class MdTabs in /home/gibson/.build/shared/js/cyph/ui/components/material/mdtabs.ts!
Cannot determine the module for class MdTextarea in /home/gibson/.build/shared/js/cyph/ui/components/material/mdtextarea.ts!
Cannot determine the module for class MdToolbar in /home/gibson/.build/shared/js/cyph/ui/components/material/mdtoolbar.ts!
Cannot determine the module for class MdTooltip in /home/gibson/.build/shared/js/cyph/ui/components/material/mdtooltip.ts!
Cannot determine the module for class Register in /home/gibson/.build/shared/js/cyph/ui/components/register.ts!
Cannot determine the module for class SignupForm in /home/gibson/.build/shared/js/cyph/ui/components/signupform.ts!
Cannot determine the module for class StaticCyphNotFound in /home/gibson/.build/shared/js/cyph/ui/components/staticcyphnotfound.ts!
Cannot determine the module for class StaticFooter in /home/gibson/.build/shared/js/cyph/ui/components/staticfooter.ts!
Cannot determine the module for class AppComponent in /home/gibson/.build/shared/js/cyph.im/appcomponent.ts!
    at analyzeAndValidateNgModules (/usr/lib/node_modules/@angular/compiler/bundles/compiler.umd.js:24187:17)
    at AotCompiler.compileAll (/usr/lib/node_modules/@angular/compiler/bundles/compiler.umd.js:23923:20)
    at CodeGenerator.codegen (/usr/lib/node_modules/@angular/compiler-cli/src/codegen.js:54:14)
    at codegen (/usr/lib/node_modules/@angular/compiler-cli/src/main.js:7:81)
    at Object.main (/usr/lib/node_modules/@angular/compiler-cli/node_modules/@angular/tsc-wrapped/src/main.js:47:16)
    at main (/usr/lib/node_modules/@angular/compiler-cli/src/main.js:13:16)
    at Object.<anonymous> (/usr/lib/node_modules/@angular/compiler-cli/src/main.js:29:5)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
Compilation failed
```

**Expected behavior**

Successful build with no errors. Everything works as expected with Angular 2.2.4 / rxjs 5.0.0-beta.12 
/ zone.js 0.7.2, but updating to Angular 2.3.0 / rxjs 5.0.0-rc.4 / zone.js 0.7.2 causes the above error.

**Minimal reproduction of the problem with instructions**

Should have some time early next week to come up with a minimal repo if needed, but this is what `AppModule` looks like:

```
import {CommonModule} from '@angular/common';
import {NgModule} from '@angular/core';
import {FormsModule} from '@angular/forms';
import {BrowserModule} from '@angular/platform-browser';
import {UpgradeModule} from '@angular/upgrade/static';
import {
	ChatCyphertext,
	ChatMain,
	ChatMessageBox,
	Checkout,
	Contact,
	FileInput,
	Help,
	LinkConnection,
	Markdown,
	Register,
	SignupForm,
	StaticCyphNotFound,
	StaticFooter
} from '../cyph/ui/components';
import {
	MdButton,
	MdCard,
	MdCardContent,
	MdCardHeader,
	MdCardHeaderText,
	MdCardTitle,
	MdCardTitleText,
	MdContent,
	MdFabSpeedDial,
	MdIcon,
	MdInput,
	MdList,
	MdListItem,
	MdMenu,
	MdOption,
	MdProgressCircular,
	MdProgressLinear,
	MdSelect,
	MdSidenav,
	MdSlider,
	MdSubheader,
	MdSwitch,
	MdTabs,
	MdTextarea,
	MdToolbar,
	MdTooltip
} from '../cyph/ui/components/material';
import {Translate} from '../cyph/ui/directives';
import {AppComponent} from './appcomponent';
import {BetaComponent} from './betacomponent';


/**
 * Angular module for Cyph UI.
 */
@NgModule({
	declarations: [
		AppComponent,
		BetaComponent,
		ChatCyphertext,
		ChatMain,
		ChatMessageBox,
		Checkout,
		Contact,
		FileInput,
		Help,
		LinkConnection,
		Markdown,
		Register,
		SignupForm,
		StaticCyphNotFound,
		StaticFooter,
		Translate,
		MdButton,
		MdCard,
		MdCardContent,
		MdCardHeader,
		MdCardHeaderText,
		MdCardTitle,
		MdCardTitleText,
		MdContent,
		MdFabSpeedDial,
		MdIcon,
		MdInput,
		MdList,
		MdListItem,
		MdMenu,
		MdOption,
		MdProgressCircular,
		MdProgressLinear,
		MdSelect,
		MdSidenav,
		MdSlider,
		MdSubheader,
		MdSwitch,
		MdTabs,
		MdTextarea,
		MdToolbar,
		MdTooltip
	],
	entryComponents: [
		AppComponent,
		FileInput,
		Help,
		Register
	],
	imports: [
		BrowserModule,
		CommonModule,
		FormsModule,
		UpgradeModule
	]
})
export class AppModule {
	/** @ignore */
	public ngDoBootstrap () : void {}

	constructor () {}
}
```

All the components starting with `Md` are upgraded ng1 components; the rest are standard ng2 components.

**What is the motivation / use case for changing the behavior?**

Using Angular in general.

**Please tell us about your environment:**

Debian Jessie inside Docker on OS X El Capitan

* **Angular version:** 2.3.0

* **Browser:** all
 
* **Language:** TypeScript 2.0.10

* **Node (for AoT issues):** 6.9.1

**Comment by vicb** [comment]
Please incude a minimal repro when you can, Thanks.

**Comment by vicb** [comment]
@chuckjaz @tbosch any idea ?

**Comment by buu700** [comment]
Was planning on setting up that minimal test tomorrow, but it looks like this issue was resolved in 2.3.1. Thanks!
